### PR TITLE
Enhance messaging setup with Factory instantiation and UI components

### PR DESCRIPTION
**Pull request by daferya**
This pull request sets up the initial messaging system, including a new MessagesScreen that uses ChatViewModel and LoggedInAccount to prepare for real-time chats between users and workers.

Right now, direct messaging between users and workers isn’t fully implemented. Instead, there’s a temporary FakeMessagesScreen that shows what the messaging screen will look like. This test screen generates sample messages directly, without using ChatViewModel, allowing a preview of the design and navigation (including back navigation)
. Related to issue #134
messagescreen

### CODE REVIEW

File: app/src/androidTest/java/com/arygm/quickfix/ui/home/MessageScreenTest.kt
Comment (mohagpt) [inline app/src/androidTest/java/com/arygm/quickfix/ui/home/MessageScreenTest.kt:1]: The test class looks solid as it covers a variety of cases and handles them well. Great job!

**Review by mohagpt — APPROVED** [review]
The new UI looks great and matches the Figma design perfectly. The code is well implemented too! For future improvements, consider using a Scaffold in the screen, this will provide a better user experience. Nice work overall!

File: app/src/main/java/com/arygm/quickfix/MainActivity.kt [hunk 1]
```diff
@@ -216,7 +219,9 @@ fun QuickFixApp() {
                 }
               }
 
-              composable(Route.HOME) { HomeNavHost(isUser) }
+              composable(Route.HOME) {
+                HomeNavHost(isUser) // , loggedInAccountViewModel, chatViewModel)
```
[the remark this answers is not in the document]
Comment (daferya) [reply app/src/main/java/com/arygm/quickfix/MainActivity.kt:223]: I prefer to keep this code commented out for now. Currently, I'm using a temporary message screen, but I’ll need this line later for the real screen. Thanks for the note!

File: app/src/main/java/com/arygm/quickfix/MainActivity.kt [hunk 2]
```diff
@@ -238,11 +243,27 @@ fun QuickFixApp() {
 }
 
 @Composable
-fun HomeNavHost(isUser: Boolean) {
+fun HomeNavHost(
+    isUser: Boolean,
+    // loggedInAccountViewModel: LoggedInAccountViewModel,
+    // chatViewModel: ChatViewModel
```
[the remark this answers is not in the document]
Comment (daferya) [reply app/src/main/java/com/arygm/quickfix/MainActivity.kt:249]: I prefer to keep this code commented out for now. Currently, I'm using a temporary message screen, but I’ll need this line later for the real screen. Thanks for the note!

File: app/src/main/java/com/arygm/quickfix/ui/home/MessagesScreen.kt [hunk 3]
```diff
@@ -1 +1,253 @@
...
+  val userId = loggedInAccount?.uid ?: return
+  val activeChatId = loggedInAccount?.activeChats?.firstOrNull()
+
+  // If no chat is active, we can display an empty interface. This will be changed later
+  if (activeChatId == null) {
+    Column(
+        modifier =
+            Modifier.fillMaxSize()
+                .background(MaterialTheme.colorScheme.background)
+                .testTag("messageScreen")) {
```
[the remark this answers is not in the document]
Comment (daferya) [reply app/src/main/java/com/arygm/quickfix/ui/home/MessagesScreen.kt:49]: That's a great idea! I'll implement that next time.

**Pull request merged**